simulator/src/resim/cmd_publish.rs

```diff
@@ -75,6 +75,7 @@ pub fn handle_publish(matches: &ArgMatches) -> Result<(), Error> {
             TransactionExecutor::new(&mut ledger, configs.current_epoch, configs.nonce);
         let transaction = TransactionBuilder::new(&executor)
             .publish_package(&code)
+            .drop_all_bucket_refs()
```

Comment:
This is an interesting side effect. If a transaction is signed but the virtual badge is never passed, users will be responsible for consuming the badge.

The good news is we're introducing auto dropping.

Reply:
Yes, I dont mind this actually as it prevents some level of transaction mutability and also helps express whether or not a transaction actually needs to be signed.